### ISSUE TITLE
hashmap erase does not take an iterator

### ISSUE THREAD

**Issue by redorav** [electronicarts/EASTL#466]
It only takes a const_iterator, but it should be able to take both.

https://en.cppreference.com/w/cpp/container/unordered_map/erase

**Comment by StArBoY-Works** [comment]
Can you explain more what exactly you want to say??

**Comment by redorav** [comment]
Yeah, what I mean is that you should be able to call myHashmap.erase(iter) with an iterator you obtained by calling find(), but instead I have to cast it to const_iter because the non-const function doesn't exist. It's a minor API annoyance but I thought I'd create an issue for it
